src/components/SwapChecker.tsx

```diff
@@ -374,5 +395,13 @@ export const SwapChecker = () => {
         }
     });
 
+    // @ts-expect-error: onbeforeunload doesn't need to return a value
+    window.onbeforeunload = (event: BeforeUnloadEvent) => {
```

Comment:
Why not return something anyways? https://developer.mozilla.org/en-US/docs/Web/API/Window/beforeunload_event